token-swap/js/client/token-swap.js

```diff
@@ -123,6 +124,13 @@ export class TokenSwap {
    */
   feeDenominator: Numberu64;
 
+  /**
+   * Curve type, current options are:
+   * 0 = Constant product curve, Uniswap-style
+   * 1 = Flat curve, always 1:1 trades
```

Comment:
Make an "enum" type for this, perhaps?

Reply:
Good idea -- I'm never sure of the best way to do JS enums though.  Do you mean something like this?
```js
const CurveType = {
    ConstantProduct: 0,
    Flat: 1
};
```

Reply:
Yep, that's what I was thinking, except:
```javascript
const CurveType = Object.freeze({
    ConstantProduct: 0,
    Flat: 1
});
```